### DISCUSSION TITLE
Does Velero support local volumes?

### DISCUSSION THREAD

**Discussion by trallnag** [vmware-tanzu/velero#3905] [Community support Q&A]
Hello,

I have read that Velero does not support hostPath. But what about local volumes?

https://kubernetes.io/docs/concepts/storage/volumes/#local
https://github.com/kubernetes-sigs/sig-storage-local-static-provisioner

**Comment by zubron** [comment]
I'm assuming you're asking in relation to restic support as that is where we explicitly mention the lack of hostPath support.

It seems like local volumes are not supported. As part of our restic restore process, we will dynamically reprovision the volume and skip the restoration of the PVs: https://github.com/vmware-tanzu/velero/blob/962a957f1b42f2568e3b89beafff6538ef42beec/pkg/restore/restore.go#L1091-L1093

This means that the PV will not be restored. As local volumes don't support dynamic provisioning, it looks like you would end up with a PVC with no associated PV to bind to. I tried this out locally, and if I recreated the PV after the restore, then the restic restore would proceed and succeed.

**Reply by zubron** [reply]
See #2520 for more details.
